### PR TITLE
Automatic export

### PR DESCRIPTION
**Pull request by ryanjob42**
Add some .xml files to make it easier to export the Jar files for the glue code

### CODE REVIEW

**Comment by lnarmour** [comment]
I like the idea, just a couple of issues:
* the xml files reference the particular versions of the jars that you happened to have in your eclipse instance at the time you committed them (they’re all named with the build num and timestamp)
* some of these may not be present (on my machine for example) but the ant build proceeds even if its can’t find all of the specified jars

When I do this:
1. Load this branch in my local eclipse instance
2. Right click `alpha.glue.v2/export.xml` > Run As > Ant Build
3. Observe that the `artifacts/bin/alpha.glue.v2.jar` file gets created successfully
```
Buildfile: /Users/lw/eclipse-cgo-artifact/alpha-codegen/bundles/alpha.glue.v2/export.xml
create_run_jar:
      [jar] Building jar: /Users/lw/eclipse-cgo-artifact/alpha-codegen/artifact/bin/alpha.glue.v2.jar
BUILD SUCCESSFUL
Total time: 2 seconds
```
4. Try to run it (via acc or however else), and see some `NoClassDefFoundError`s:
```
acc -o out test.alpha 
[acc]: reading 'test.alpha' file
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.jdt.internal.jarinjarloader.JarRsrcLoader.main(JarRsrcLoader.java:61)
Caused by: java.lang.NoClassDefFoundError: org/apache/log4j/Logger
	at org.eclipse.xtext.service.AbstractGenericModule.<clinit>(AbstractGenericModule.java:29)
	at org.eclipse.xtext.common.TerminalsStandaloneSetupGenerated.createInjector(TerminalsStandaloneSetupGenerated.java:45)
	at org.eclipse.xtext.common.TerminalsStandaloneSetupGenerated.createInjectorAndDoEMFRegistration(TerminalsStandaloneSetupGenerated.java:39)
	at org.eclipse.xtext.common.TerminalsStandaloneSetup.doSetup(TerminalsStandaloneSetup.java:18)
	at alpha.model.AlphaStandaloneSetupGenerated.createInjectorAndDoEMFRegistration(AlphaStandaloneSetupGenerated.java:19)
	at alpha.model.AlphaStandaloneSetup.doSetup(AlphaStandaloneSetup.java:13)
	at alpha.targetmapping.TargetMappingStandaloneSetupGenerated.createInjectorAndDoEMFRegistration(TargetMappingStandaloneSetupGenerated.java:19)
	at alpha.loader.AlphaLoader.initialize(AlphaLoader.java:61)
	at alpha.loader.AlphaLoader.<clinit>(AlphaLoader.java:57)
	at alpha.glue.v2.GenerateNewWriteC.main(GenerateNewWriteC.java:40)
	... 5 more
Caused by: java.lang.ClassNotFoundException: org.apache.log4j.Logger
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 15 more
```


The problem here is that the v2 export file references `org.apache.log4j_1.2.25.jar` but I happen to have a different version installed in my Eclipse instance:
```
$ ls Eclipse.app/Contents/Eclipse/plugins | grep apache.log4j
org.apache.log4j_1.2.19.v20220208-1728.jar
```

And apparently ant just skips this, I can see that there is no log4j jar in the exported `alpha.glue.v2.jar` file:
```
$ jar -tf artifact/bin/alpha.glue.v2.jar | grep log4j
$
```

Can we wildcard the build timestamps and/or make ant fail if it can’t find everything?

**Comment by lnarmour** [comment]
Very nice, the alpha.glue.v1 project exports. Ant will still run (and appear successful) if one or more of the jars does not exist though. It looks like ant can fail conditionally, but it has to be told:
https://ant.apache.org/manual/Tasks/fail.html

I got it to work by adding a new target with a fail task per jar that fails if the jar doesn’t exist. Then I made the create_run_jar target depend on this. I pushed it to the `automatic-export-conditional-fail` branch. See commit https://github.com/CSU-CS-Melange/alpha-codegen/commit/1573e5566254c93493cffb2748fd74d139c19ea5 for a diff.

Here’s a bash script/function that you could use to generate the fail task
```bash
function make_fail_task {
JAR_STR=$1
cat <<-EOF
<fail message="$JAR_STR is missing">
  <condition>
    <not>
      <resourcecount count="1">
        <fileset dir="\${ant.home}/../" includes="$JAR_STR"/>
      </resourcecount>
    </not>
  </condition>
</fail> 
EOF
}
```

File: bundles/alpha.glue.v2/export.xml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            </manifest>
+            <zipfileset src="jar-in-jar-loader.zip"/>
+            <fileset dir="${dir.buildfile}/bin"/>
+            <fileset dir="${basedir}../../../alpha-language/bundles/alpha.codegen/bin"/>
+            <fileset dir="${basedir}../../../alpha-language/bundles/alpha.model/bin"/>
+            <mappedresources>
+                <zipfileset dir="${ant.home}/../">
+                	<include name="org.eclipse.emf.ecore_2.27.0.v20220426-0617.jar"/>
```
Comment (lnarmour) [inline bundles/alpha.glue.v2/export.xml]: All of the v2 exports still have the build data/time. I’m guessing you were waiting to see if the v1 updates works.

Comment (ryanjob42) [reply bundles/alpha.glue.v2/export.xml]: Ah, no I just missed that somehow. Fixed now

**Comment by ryanjob42** [comment]
Fail tasks have been added

**Pull request merged**